src/analysis/special_functions/bernstein.lean

```diff
@@ -0,0 +1,378 @@
+  such a coin gives exactly `k` heads in a sequence of `n` tosses.
+* If such an appearance of `k` heads results in a payoff of `f(k / n)`,
+  the `n`-th Bernstein approximation for `f` evaluated at `x` is the expected payoff.
+* The main estimate in the proof bounds the probability that
+  the observed frequency of heads differs from `x` by more than some `δ`,
+  obtaining a bound of `(4 * n * δ^2)⁻¹`, irrespective of `x`.
+* This ensures that for `n` large, the Bernstein approximation is (uniformly) close to the
+  payoff function `f`.
+
+(You don't need to think in these terms to follow the proof below: it's a giant `calc` block!)
+
+This result proves Weierstrass' theorem that polynomials are dense in `C([0,1], ℝ)`,
+although we defer an abstract statement of this until later.
+-/
+
+
+noncomputable theory
+
+open_locale classical
+open_locale big_operators
+open_locale bounded_continuous_function
+
+-- FIXME find a home for this
+lemma mul_unit_interval_le {α : Type*} [ordered_semiring α] {a b c : α}
+  (h₁ : 0 ≤ c) (h₂ : a ≤ c) (h₃ : 0 ≤ b) (h₄ : b ≤ 1) : a * b ≤ c :=
+begin
+  conv_rhs { rw ←mul_one c, },
+  exact mul_le_mul h₂ h₄ h₃ h₁,
+end
+
+/--
+The Bernstein polynomials, as continuous functions on ℝ.
+-/
+def bernstein' (n ν : ℕ) : C(ℝ, ℝ) :=
+⟨λ x : ℝ, (bernstein_polynomial ℝ n ν).eval x, by continuity⟩
+
+-- TODO there's some orphaned development regarding `[0,1]` in `topology.path_connected`;
+-- perhaps this should be consolidated.
+local notation `I` := set.Icc (0 : ℝ) (1 : ℝ)
+
+namespace unit_interval
+
+lemma nonneg (x : I) : 0 ≤ (x : ℝ) := x.2.1
+lemma one_minus_nonneg (x : I) : 0 ≤ 1 - (x : ℝ) := by simpa using x.2.2
+lemma le_one (x : I) : (x : ℝ) ≤ 1 := x.2.2
+lemma one_minus_le_one (x : I) : 1 - (x : ℝ) ≤ 1 := by simpa using x.2.1
+
+end unit_interval
+
+namespace tactic.interactive
+
+/-- A tactic that solves `0 ≤ x`, `0 ≤ 1 - x`, `x ≤ 1`, and `1 - x ≤ 1` for `x : I`. -/
+meta def unit_interval : tactic unit :=
+`[apply unit_interval.nonneg] <|> `[apply unit_interval.one_minus_nonneg] <|>
+`[apply unit_interval.le_one] <|> `[apply unit_interval.one_minus_le_one]
+
+end tactic.interactive
+
+instance : has_zero I := ⟨⟨0, ⟨le_refl _, le_of_lt real.zero_lt_one⟩⟩⟩
+instance : has_one I := ⟨⟨1, ⟨le_of_lt real.zero_lt_one, le_refl _⟩⟩⟩
+instance : nonempty I := ⟨0⟩
+
+-- FIXME Where do these lemmas belong?
+-- Should they become part of a public API, or remain hidden here?
+-- They really belong on `C(α,β)` rather than `α →ᵇ β`.
+
+namespace bounded_continuous_function
+variables {α β : Type*} [metric_space α] [compact_space α] [metric_space β]
+
+/-!
+We now set up some abbreviations for the various components of
+uniform continuity for a continuous function on a compact metric space.
+-/
+lemma uniform_continuity
+  (f : α →ᵇ β) (ε : ℝ) (h : 0 < ε) :
+  ∃ δ > 0, ∀ {x y}, dist x y < δ → dist (f x) (f y) < ε :=
+metric.uniform_continuous_iff.mp
+  (compact_space.uniform_continuous_of_continuous f.2.1) ε h
```

Comment:
This one definitely deserves to live publicly in `C(α,β)`!  Not sure about the others.